cmd/serverless/api_key.go

```diff
@@ -102,12 +106,15 @@ func readAPIKeyFromSecretsManager(arn string) (string, error) {
 		return "", err
 	}
 
-	sess, err := session.NewSession(nil)
+	sess, err := session.NewSession(aws.NewConfig().WithHTTPClient(&http.Client{
+		Transport: datadogHttp.CreateHTTPTransport(),
+	}))
```

Comment:
Difficult to add tests, and `CreateHTTPTransport` is already tested so I believe it would be redundant.